### ISSUE TITLE
Pixel scrolling using the keyboard?

### ISSUE THREAD

**Issue by NightMachinery** [ahrm/sioyek#958]
When scrolling using the keyboard, e.g.,
```
move_up k
move_down j
```
, the scrolling happens in discrete steps when one holds down the key `j`. This looks much worse than continuous/animated scrolling which is possible with the mouse/trackpad. (Emacs calls this feature "pixel scrolling.")

**Comment by ahrm** [comment]
Added `move_down_smooth` and `move_up_smooth` commands in https://github.com/ahrm/sioyek/commit/1bf37bdb697b7ca83e14a284833012809a06f9bd.

**Comment by NightMachinery** [comment]
Thanks. I tested these two, and they work. I initially wanted a way to set the initial velocity, but I worked around it by using:
```
move_up_smooth;move_up_smooth k
move_down_smooth;move_down_smooth j
```

Is there a way to make commands such as `screen_down` move using an animation?

I generally dislike animations in the UI, but for scrolling, it can be helpful; it shows you exactly how much scrolling was done and in which directions, and allows the eyes to more easily anchor themselves to the "current" section they were looking at previously. 

I am also thinking of how such animated move commands could be composed (like in my `next_column_smart` commands which combined several move commands) ...

**Comment by NightMachinery** [comment]
Can we make the movement stop immediately when the key is released? (I.e., no dampening, just a complete stop) Perhaps with another config option that controls the dampening strength.

**Comment by ahrm** [comment]
> it shows you exactly how much scrolling was done and in which directions, and allows the eyes to more easily anchor themselves to the "current" section they were looking at previously.

I recommend using the ruler for this, you can create a ruler under lines by right clicking and then up/down keys move this ruler, also you can use `toggle_visual_scroll` to use mouse wheel to control the ruler.

**Comment by NightMachinery** [comment]
I tested the ruler, it doesn't seem useful to me. The ruler moves in single lines using the keyboard, and doesn't work with `screen_up` etc. I generally don't want to use the mouse at all, as I have a disability in my hands which make a keyboard-only workflow much easier on me.

**Comment by ahrm** [comment]
Added in https://github.com/ahrm/sioyek/commit/1bbf24b0964c718f4f2ffe68690fa2e1a73bd751. We now have two config options: `smooth_move_initial_velocity` and `smooth_move_max_velocity`. When the key is pressed only once we set the speed to initial velocity and smoothly move until the velocity reaches zero. When the key is held, we increase the velocity until we reach max velocity and when the key is lifted we immediately stop. Also we now have ability to define holdable commands. For example in `keys_user.config`:
```
{holdable}toggle_dark_mode|toggle_dark_mode z
```
The command should starts with `{holdable}` and be followed by two `|` separated commands, the first is executed when a key is pressed and the other when it is lifted.
